### PR TITLE
WIP: Changing Json to protobufs for login crate

### LINKED ISSUE #1605: The logins crate should use protobufs
The following methods return their results as JSON because we `implement_into_ffi_by_json!(Login);`
https://github.com/mozilla/application-services/blob/fe9c866787898d40fae1294d218a211c0b8c0c15/components/logins/ffi/src/lib.rs#L166-L201

The work to do is [the same as what we've done on the Push crate](https://github.com/mozilla/application-services/issues/1491#issue-474206810), plus:

* Swift support ([example](https://github.com/mozilla/application-services/blob/4885cd11fcbacebaa707f7d78878ca15918b7c38/components/places/ios/Places/Places.swift#L331-L345))
* (Optional/Good follow-up) `sync15_passwords_add` also takes JSON as input! What if we could take protobuf as an input instead? We do this for the places crate in [Kotlin](https://github.com/mozilla/application-services/blob/4885cd11fcbacebaa707f7d78878ca15918b7c38/components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt#L488-L495) and [Swift](https://github.com/mozilla/application-services/blob/4885cd11fcbacebaa707f7d78878ca15918b7c38/components/places/ios/Places/Places.swift#L777-L787).
* Finally `implement_into_ffi_by_json!(Login);` can be deleted.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-743)

### PR DESCRIPTION
**Pull request by schoenenberg**
Tries to fix #1605. Continue the work of @tinoism in PR #1671.

### CODE REVIEW

File: components/logins/build.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+fn main() {
+    println!("cargo:rerun-if-changed=src/places_msg_types.proto");
```
Comment (thomcc) [inline components/logins/build.rs]: This file is wrong. You should be adding a new message type file to `src/logins_msg_types.proto` (unfortunately the `logins_` name prefix is required due to some idiosyncracies in the iOS build process), and not referring to the places types and etc.

File: components/logins/src/ffi.rs [hunk 2]
```diff
@@ -111,4 +116,35 @@ impl From<Error> for ExternError {
     }
 }
 
-implement_into_ffi_by_json!(Login);
+impl From<Login> for msg_types::Login {
+    fn from(login: Login) -> Self {
+        Self {
+            guid: login.guid,
+
```
Comment (thomcc) [inline components/logins/src/ffi.rs]: Nit: This shouldn't have a newline between each line.

File: components/logins/build.rs (hunk 1 above)
Comment (schoenenberg) [reply components/logins/build.rs]: Thanks, I have seen the comment on top of the `.proto` file.

**Comment by schoenenberg** [comment]
Unfortunately, I'm new to this project and don't know the hurdles of this build pipeline. I need some help with this build issue:
```
error[E0463]: can't find crate for `_bytes`
 --> /home/circleci/project/target/debug/build/logins-a8d16c9c8297dc35/out/msg_types.rs:1:28
  |
1 | #[derive(Clone, PartialEq, ::prost::Message)]
  |                            ^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error
```

**Comment by schoenenberg** [comment]
> Unfortunately, I'm new to this project and don't know the hurdles of this build pipeline. I need some help with this build issue:
> 
> ```
> error[E0463]: can't find crate for `_bytes`
>  --> /home/circleci/project/target/debug/build/logins-a8d16c9c8297dc35/out/msg_types.rs:1:28
>   |
> 1 | #[derive(Clone, PartialEq, ::prost::Message)]
>   |                            ^^^^^^^^^^^^^^^^ can't find crate
> 
> error: aborting due to previous error
> ```

Crate "bytes" was missing in `Cargo.toml`. See https://github.com/mozilla/application-services/pull/1671#issuecomment-525119711

**Comment by schoenenberg** [comment]
This test case from `db.rs` is failing:
```
[2019-11-26T00:09:31Z ERROR logins::db] Failed to deserialize record Guid("dummy_000002"): missing field `hostname`
```
Isn't that correct behavior? It's catched as expected by `fetch_login_data()`

**Comment by eoger** [comment]
What's the status on this PR? Is there action needed on our side?

**Pull request closed without merging**